src/models/shows.js

```diff
@@ -0,0 +1,140 @@
+import Joi from 'joi';
+import Boom from 'boom';
+
+const showSchema = Joi.object().keys({
+    _id: Joi.string(),
+    showName: Joi.string().required(),
+    users: Joi.array().items(Joi.string()).required(),
+    dayOfWeek: Joi.string().required(),
+    startTime: Joi.number().integer().required(),
+    endTime: Joi.number().integer().required(),
+    isActive: Joi.boolean().required()
+});
+
+
+const getShows = (request, reply) => {
+    const db = request.server.plugins['hapi-mongodb'].db;
+    db.collection('shows').find({}, async (err, cursor) => {
+        if (err) {
+            return reply(Boom.internal('Internal MongoDB error', err));
+        }
+
+        const shows = await cursor.toArray();
+
+        return reply(shows);
+    });
+};
+
+const updateShow = (request, reply) => {
+    const { db, ObjectID } = request.server.plugins['hapi-mongodb'];
+    const show = request.payload;
+
+    try {
+        Joi.validate(show, showSchema, (err, value) => {
```

Comment:
I'll leave it for now, since I'm fairly deep in this pattern. I opted for it largely to be a bit more flexible in their placement within a route handler, but we can always add additional route validations to the route itself later!